src/main/java/de/dennisguse/opentracks/io/file/importer/AbstractFileTrackImporter.java

```diff
@@ -279,17 +255,26 @@ protected void onTrackEnd() {
         }
         trackData.track.setTrackStatistics(trackData.trackStatisticsUpdater.getTrackStatistics());
 
-        try {
-            contentProviderUtils.updateTrack(trackData.track);
-        } catch (SQLiteConstraintException e) {
+        Track track = contentProviderUtils.getTrack(trackData.track.getUuid());
+        if (track != null) {
             if (PreferencesUtils.getPreventReimportTracks(context)) {
-                throw e;
+                throw new ImportAlreadyExistsException(new SQLiteConstraintException());
```

Comment:
I wouldn't encapsulate SQLiteConstraintException, because it didn't happen here.

Reply:
Thank you @pstorch for your review. What do you think something of this:

```java
throw new ImportAlreadyExistsException(context.getString(R.string.import_prevent_reimport));
```

with R.string.import_prevent_reimport = "You have set prevent re-import track"

Reply:
Normally I don't like translating Exception messages, but in this case we show it to the user. So it might be appropriate.
@dennisguse what do you think?